data-processing/entities/definitions/nlp.py

```diff
@@ -62,200 +61,197 @@ def __init__(self, prediction_types: List[str]) -> None:
+                "Could not load model from %s. A pre-trained model could "
+                + "not be found in the directory. This can occur if the download of the model was "
+                + "terminated. Try deleting %s and running this script again.",
                 training_args.output_dir,
+                cache_directory,
+            )
+            raise ValueError(
+                f"Could not load model from {training_args.output_dir}"
             )
-
-            # Load the model.
-            model_class = MODEL_CLASSES[model_args.model_type]
-            if (
-                os.path.exists(training_args.output_dir)
-                and not training_args.overwrite_output_dir
-            ):
-                model = model_class.from_pretrained(
-                    training_args.output_dir,
-                    args=training_args,
-                    intent_label_dict=get_intent_labels_dict(data_args),
-                    slot_label_dict=get_slot_labels_dict(data_args),
-                    pos_label_lst=get_pos_labels(data_args),
-                    # This is because currently there are 3 different models - one for each task
-                    tasks=[prediction_type],
-                )
-                logging.info("Model loaded from %s", training_args.output_dir)
-            else:
-                logging.error(  # pylint: disable=logging-not-lazy
-                    "Could not load model from %s. A pre-trained model could "
-                    + "not be found in the directory. This can occur if the download of the model was "
-                    + "terminated. Try deleting %s and running this script again.",
-                    training_args.output_dir,
-                    cache_directory,
-                )
-                raise ValueError(
-                    f"Could not load model from {training_args.output_dir}"
-                )
 
             # model.resize_token_embeddings(len(tokenizer))
 
-            data_args.ignore_index = training_args.ignore_index
-            self.data_args[prediction_type] = data_args
-            self.model_args[prediction_type] = model_args
-
-            self.tokenizer[prediction_type] = tokenizer
-            self.model[prediction_type] = model
-            self.trainer[prediction_type] = Trainer(
-                [
-                    training_args,
-                    self.model_args[prediction_type],
-                    self.data_args[prediction_type],
-                ],
-                self.model[prediction_type],
-            )
+        data_args.ignore_index = training_args.ignore_index
+        self.data_args = data_args
+        self.model_args = model_args
+
+        self.tokenizer = tokenizer
+        self.model = model
+        self.trainer = Trainer(
+            [
+                training_args,
+                self.model_args,
+                self.data_args,
+            ],
+            self.model,
+        )
 
```

Comment:
@andrewhead  Every instance where we had a dict of 3 models, 3 tokenizers etc have been replaced by one single model, tokenizer etc.

Reply:
sounds good!